### ISSUE TITLE
Consistent mapply error on read_srt

### ISSUE THREAD

**Issue by asitemade4u** [kiernann/srt#3]
Hi,
I am treating a lot of subtitles to create a text corpus and face the same issue with srt again and again: 
`Error in mapply(FUN = f, ..., SIMPLIFY = FALSE) : 
  zero-length inputs cannot be mixed with those of non-zero length`

I attach a sample `.srt` file that does not go thru, for you to test: 
[Barefoot in the Park (1967) 1080p AAC.eng.zip](https://github.com/kiernann/srt/files/7086049/Barefoot.in.the.Park.1967.1080p.AAC.eng.zip)

Best,
Stephen

PS. I am using the CRAN version of the plugin because I could not succeed in installing it using `remotes::install_github("kiernann/srt")`, and I acknowledge that might be the reason

**Comment by k5cents** [comment]
This problem is stemming from the 3 blank lines at the end of your file. The SubRip standard uses newlines (`\n`) to separate subtitle lines, so the package is looking for another subtitle after line 9617.

|    n|line                          |
|----:|:-----------------------------|
| 9614|01:45:54,765 --> 01:45:57,234 |
| 9615|* Knee-deep in daisies        |
| 9616|with you *                    |
| 9617|                              |
| 9618|                              |
| 9619|                              |

I'll work on a consistent fix, but in the meantime you can use `readLines()` and then trim any blank lines at the end before overwriting with `writeLines()`.

**Comment by asitemade4u** [comment]
WOW, thanks for responding so fast! Will test it right away.

**Comment by k5cents** [comment]
I've added some code to remove consecutive blank lines before parsing the text for times, etc. This should fix your issue in the most recent development version. If issue #4 is still bothering you, this fix should be coming to `1.0.3` which I will send to CRAN tomorrow.

``` r
library(srt)
packageVersion("srt")
#> [1] '1.0.2.9001'

download.file(
  url = "https://github.com/kiernann/srt/files/7086049/Barefoot.in.the.Park.1967.1080p.AAC.eng.zip",
  destfile = tmp <- tempfile()
)

srt <- unzip(tmp, exdir = tempdir())
out <- read_srt(srt)
tail(out)
#> # A tibble: 6 × 4
#>       n start   end subtitle                                 
#>   <int> <dbl> <dbl> <chr>                                    
#> 1  2018 6322. 6324. "Oh, good.\nThey made up."               
#> 2  2019 6333. 6336. "* La la la..."                          
#> 3  2020 6341. 6345. "* So come along My barefoot love"       
#> 4  2021 6345. 6348. "* To the fields\nthat shine with spring"
#> 5  2022 6349. 6353. "* Though it's 10 below Let me go"       
#> 6  2023 6355. 6357. "* Knee-deep in daisies\nwith you *"
```

<sup>Created on 2021-08-31 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

**Comment by k5cents** [comment]
@asitemade4u srt 1.0.3 is live on CRAN and should fix the example problem in your issue.
